modules/assignment/src/main/java/com/intuit/wasabi/assignment/AssignmentsModule.java

```diff
@@ -100,6 +111,42 @@ protected void configure() {
         LOGGER.debug("installed module: {}", AssignmentsModule.class.getSimpleName());
     }
 
+    private void bindMetadataCache(final Properties properties) {
+        Boolean metadataCacheEnabled = Boolean.parseBoolean(getProperty("metadata.cache.enabled", properties, "true"));
+        bind(Boolean.class).annotatedWith(named("AssignmentsMetadataCacheEnabled")).toInstance(metadataCacheEnabled);
+
+        if(metadataCacheEnabled) {
+            //This is a cache refresh interval, at this frequency cache will be refreshed.
+            Integer metadataCacheRefreshInterval = Integer.parseInt(getProperty("metadata.cache.refresh.interval", properties, "5"));
+            Integer metadataCacheNumberOfThreads = 1; //We want only single thread to refresh metadata cache.
```

Comment:
ScheduledExecutorService takes care of choosing a thread to run/execute AssignmentsMetadataCacheRefreshTask.

So, in case if current thread dies then it should create new one and use to run/execute AssignmentsMetadataCacheRefreshTask.